woocommerce/wc-admin
# Dashboard: Add Leaderboard Component

For #1192

This branch introduces a new `<Leaderboard />` component that is heavily inspired by the `<ReportTable />` component. Initially I started to add a prop to re-use all of the table.js logic for a new Categories Leaderboard block, but that seemed to get a bit too complicated - as the tables on the report pages are much more full featured ( filtering, download button, compare checkbox etc ).

So I opted to make a similar component just for Dashboard leaderboards... and here we are!

In this branch I have implemented the `Leaderboard` component, and updated the "Top Selling Products" leaderboard to use the new component. While doing so the following little buggers were fixed:

- Migrated Top Selling Products to use the `wc-api` store / fresh data.
- Fixed the Product name not showing up within the Top Selling Products leaderboard
- Top Selling Products now respects/uses the Date Picker range selected on the Dashboard

### Accessibility




### Screenshots

![top-selling-branch](https://user-images.githubusercontent.com/22080/50501323-35340e00-0a0c-11e9-94ff-d837421b4bd3.png)

### Detailed test instructions:

- Open the Dashboard and verify the Top Selling Products leaderboard displays properly ( loading indicator / rendering etc )
- Change the date range using the Date Picker - verify the change triggers a refresh of the Top Selling Products leaderboard
